src/io/flutter/server/vmService/VMServiceManager.java

```diff
@@ -249,12 +251,24 @@ else if (!stream.getValue()) {
         stream.setValue(true);
       }
 
-      if (serviceExtensionState.get(name) != null && serviceExtensionState.get(name).getValue()) {
-        // We should not call the service extension for the follwing extension.
-        if (StringUtil.equals(name, "ext.flutter.inspector.show")) {
-          return;
+      // Restore any previously true states by calling their service extensions.
+      if (getServiceExtensionState(name).getValue()) {
+        restoreServiceExtensionState(name);
+      }
+    }
+  }
+
+  private void restoreServiceExtensionState(String name) {
+    if (app.isSessionActive()) {
+      // We should not call the service extension for the follwing extensions.
+      if (!StringUtil.equals(name, "ext.flutter.inspector.show")
```

Comment:
negated logic and nested ifs are harder to read than repeated if else cases.
Rewrite as

if (StringUtil.equals(name, "ext.flutter.inspector.show") || 
 StringUtil.equals(name, "ext.flutter.platformOverride") {
  // Do not call service extension for these extensions because...
} else if (StringUtil.equals(name, "ext.flutter.timeDilation")) {
   final Map<String, Object> params = new HashMap<>();
          params.put("timeDilation", 5.0);
          app.callServiceExtension("ext.flutter.timeDilation", params);
} else {
  app.callBooleanExtension(name, true);
}

Reply:
Fixed.